lib/datadog/appsec/processor.rb

```diff
@@ -16,9 +16,38 @@ def self.===(other)
         end
       end
 
+      # Context wraps libddwaf's context
+      class Context
+        attr_reader :time, :time_ext, :timeouts
+
+        def initialize(processor)
+          @context = Datadog::AppSec::WAF::Context.new(processor.send(:handle))
+          @time = 0.0
+          @time_ext = 0.0
+          @timeouts = 0
+        end
+
+        def run(*args)
+          start = Core::Utils::Time.get_time
+
+          ret, res = @context.run(*args)
+
+          stop = Core::Utils::Time.get_time
+
+          @time += res.total_runtime
+          @time_ext += (stop - start) * 1_000_000_000
+          @timeouts += 1 if res.timeout
+
+          [ret, res]
+        end
+      end
+
+      attr_reader :ruleset_info, :addresses
+
       def initialize
         @ruleset = nil
         @handle = nil
+        @ruleset_info = nil
 
```

Comment:
Minor: Should `@addresses` also be initialized here to `nil`?

Reply:
One is not really supposed to query `addresses` if `create_waf_handle` failed, but for the sake of consistency, sure.